splunk-quarkus/src/main/java/com/redhat/console/integrations/BasicAuthenticationProcessor.java

```diff
@@ -0,0 +1,48 @@
+package com.redhat.console.integrations;
+
+import java.nio.charset.StandardCharsets;
+
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.Processor;
+import org.apache.camel.util.json.JsonObject;
+import org.apache.http.Header;
+// Import httpclient classes.
+// Currently form version 4, which depends on camel-http package.
+import org.apache.http.auth.Credentials;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.impl.auth.BasicScheme;
+
+/*
+ * Sets Authorization header as a Basic authentication
+ * reading the password from X-Insight-Token metadata.
+ *
+ * Camel HTTP supports basic auth, however it cannot be dynamically
+ * set, unless toD (dynamic) would be used.
+ */
+public class BasicAuthenticationProcessor implements Processor {
+
+    private String username;
+
+    public BasicAuthenticationProcessor(String username) {
+        this.username = username;
+    }
+
+    public void process(Exchange exchange) {
+        Message in = exchange.getIn();
+
+        JsonObject metadata = in.getHeader("metadata", JsonObject.class);
+        String password = metadata.getString("X-Insight-Token");
+        if (password == null || password.equals("")) {
+            return;
+        }
+
+        Credentials credentials = new UsernamePasswordCredentials(this.username, password);
+
+        BasicScheme basicScheme = new BasicScheme();
+        Header header = basicScheme.authenticate(credentials, StandardCharsets.US_ASCII.name(), false /* proxy **/);
```

Comment:
This one particular method is deprecated - https://github.com/apache/httpcomponents-client/blob/4.5.x/httpclient/src/main/java/org/apache/http/impl/auth/BasicScheme.java#L194

Reply:
Changed to use the non-deprecated method. Created `BasicHttpRequest` and `BasicHttpContext` for it.